src/game.cpp

```diff
@@ -10872,24 +10872,45 @@ void game::chat()
     }
 
     uimenu nmenu;
-    nmenu.text = std::string( _("Who do you want to talk to?") );
-
+    nmenu.text = std::string( _( "Who do you want to talk to or yell at?" ) );
+    
     int i = 0;
+
     for( auto &elem : available ) {
         nmenu.addentry( i++, true, MENU_AUTOASSIGN, ( elem )->name );
     }
 
     nmenu.return_invalid = true;
-    nmenu.addentry( i++, true, 'a', _( "Yell" ) );
-    nmenu.addentry( i++, true, 'q', _( "Cancel" ) );
+
+    int yell, yell_sentence, cancel;
+
+    nmenu.addentry( yell = i++, true, 'a', _( "Yell" ) );
```

Comment:
> An `enum` could have been used here.

We discussed the solution you propose with @AlecWhite before and came to conclusion that an `enum` wouldn't work here. If choice numbers from 0 to n - 1 are occupied for selecting NPCs and you want the `yell` choice to be n, there's no way to turn that n into an enum constant, because n is variable.

> This kind of assignment is ugly and confusing.

Not as much as this statement of yours, fagget.